lib/nx_map.c

```diff
@@ -0,0 +1,191 @@
+#include <stdlib.h>
```

Comment:
I expected to see a copyright header here mentioning the 2 licenses of this file/project.